src/TouchCarousel.js

```diff
@@ -55,16 +55,35 @@ class TouchCarousel extends React.PureComponent {
     this.grabbing = false
     this.tracingTouchId = null
     this.isMovingCross = null
+    this.isMouseDown = false
+    this.mouseTouchId = 0
   }
 
   componentDidMount () {
+    document.addEventListener('mousemove', this.onDocumentMouseMove)
+    document.addEventListener('mouseup', this.onDocumentMouseUp)
+    document.addEventListener('blur', this.onDocumentMouseUp)
     this.autoplayIfEnabled()
   }
 
   componentWillUnmount () {
+    document.removeEventListener('mousemove', this.onDocumentMouseMove)
+    document.removeEventListener('mouseup', this.onDocumentMouseUp)
+    document.removeEventListener('blur', this.onDocumentMouseUp)
     this.stopAutoplay()
   }
 
+  mockTouchEvent = (e) => {
+    e.changedTouches = [{ identifier: this.mouseTouchId, pageX: e.pageX, pageY: e.pageY }]
+    return e
```

Comment:
Does a `blur` event has sane pageX and pageY?

And I think we'd better not modify the original events. Returning a plain object is good enough.
 
Also, `e.type` matters. It is used to choose which callback we call, onDragEnd or onDragCancel. See `onTouchEndOrCancel()`